sdk/cosmos/azure-cosmos/test/crud_tests.py

```diff
@@ -859,7 +860,8 @@ def test_document_crud(self):
 
         # should pass for most recent etag
         replaced_document_conditional = created_collection.replace_item(
```

Comment:
We should probably add some tests to cover the new errors being thrown when match_condition is specified without etag and other cases.

Reply:
Added :)